### PR TITLE
Arm interface

### PR DESCRIPTION
**Pull request by ennerf**
This PR provides a higher-level abstraction for working with (serial chain) arms. The main features are:

* moving along trajectories with gravity compensation and pos/vel/effort control
* seamless replanning of trajectories
* keeping track of auxiliary (e.g. gripper) state
* a plugin system that allows adding e.g. impedance control as needed

This PR also includes changes to the API itself, which will be released separately if this PR gets approved
* `minMoveTime` now applies in between multiple waypoints and not only on the entire duration (this behavior was intended for 2 waypoints, so this can be considered a bugfix)
* `HebiUtils.sendWithRetry`

### CODE REVIEW

**Comment by ennerf** [comment]
I updated it to the official 1.7.1 matlab release, so I'm ok merging it

File: include/input/HebiKeyboard.m [hunk 1]
```diff
@@ -85,9 +87,27 @@ function loadLibs()
             
         end
         
-        function out = read(this)
+        function [keys, diffKeys] = read(this)
```
Comment (iamtesch) [inline include/input/HebiKeyboard.m]: This MATLAB feature is super nice for backwards compatibility...

File: kits/arms/experimental/HebiArm.m [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+            this.state = newState;
+            for i=1:length(this.plugins)
+                this.plugins{i}.update(this);
+            end            
+            
+            % TODO: --> user code
+            % * add effort offsets for e.g. spring.
+            % * add efforts to add an end effector wrench (local or world coordinates?)
+            % * add impedance controller
+            % * remove efforts if all zeros? e.g. strategy 3
```
Comment (iamtesch) [inline kits/arms/experimental/HebiArm.m]: The impedance controller and effort offsets are already done, right?  Should we take out these TODOs?

File: include/input/HebiKeyboard.m (hunk 1 above)
Comment (ennerf) [reply include/input/HebiKeyboard.m]: I agree 👍

File: kits/arms/experimental/HebiArm.m (hunk 2 above)
Comment (ennerf) [reply kits/arms/experimental/HebiArm.m]: an oversight. thanks

**Comment by ennerf** [comment]
I created a new namespace, so plugins show up in autocomplete now, e.g.,

plugin = HebiArmPlugins.EffortOffset(offsets)

**Pull request merged**